modules/openapi-generator/src/main/resources/ruby-client/partial_oneof_module.mustache

```diff
@@ -0,0 +1,69 @@
+  {{#description}}
+  # {{{description}}}
+  {{/description}}
+  module {{classname}}
+    {{#oneOf}}
+    {{#-first}}
+    # List of class defined in oneOf (OpenAPI v3)
+    def self.openapi_one_of
+      [
+    {{/-first}}
+        :'{{{.}}}'{{^-last}},{{/-last}}
+    {{#-last}}
+      ]
+    end
+
+    {{/-last}}
+    {{/oneOf}}
+    {{#discriminator}}
+    {{#propertyName}}
+    # discriminator's property name in OpenAPI v3
+    def self.openapi_discriminator_name
+      :'{{{.}}}'
+    end
+
+    {{/propertyName}}
+    {{#mappedModels}}
+    {{#-first}}
+    # discriminator's mapping in OpenAPI v3
+    def self.openapi_discriminator_mapping
+      {
+    {{/-first}}
+        :'{{{mappingName}}}' => :'{{{modelName}}}'{{^-last}},{{/-last}}
+    {{#-last}}
+      }
+    end
+    {{/-last}}
+    {{/mappedModels}}
+
+    {{/discriminator}}
+    # Builds the object from hash
+    # @param [Hash] attributes Model attributes in the form of hash
+    # @return [Object] Returns the model itself
+    def self.build_from_hash(attributes)
+    {{#discriminator}}
+      discriminator_value = attributes[openapi_discriminator_name]
+      return nil unless discriminator_value
+    {{#mappedModels}}
+    {{#-first}}
+
+      _class = openapi_discriminator_mapping[discriminator_value.to_sym]
+      return nil unless _class
+
+      {{moduleName}}.const_get(_class).build_from_hash(attributes)
+    {{/-first}}
+    {{/mappedModels}}
+    {{^mappedModels}}
+      {{moduleName}}.const_get(discriminator_value).build_from_hash(attributes)
+    {{/mappedModels}}
+    {{/discriminator}}
+    {{^discriminator}}
+      openapi_one_of.each do |_class|
+        model = {{moduleName}}.const_get(_class).build_from_hash(attributes)
```

Comment:
This should be wrapped in a try/rescue. `build_from_hash` for enum values can raise. And when that's the case, the loop should continue to try the next model.

Reply:
> build_from_hash for enum values can raise

can you provide an example?

Reply:
It's in here https://github.com/OpenAPITools/openapi-generator/blob/a0a7464786eb727d1ca4b483ad711bc79bc2bdf6/modules/openapi-generator/src/main/resources/ruby-client/partial_model_enum_class.mustache#L17

Reply:
Oh one more thing I noticed, as I'm trying out this PR on our spec: the `oneOf` types can be primitive types like `Float`, `String`, ... so we can't call `build_from_hash` on those. Maybe you could add some special cases, comparing the `openapi_one_of` value to primitive types, and try to directly parse the value in that case ?

Reply:
Yes, this has been pointed out and it is something I plan to address https://github.com/OpenAPITools/openapi-generator/pull/5706#issuecomment-695869515

Reply:
Ah right, sorry for the noise 🙈